app/views/grade_entry_forms/_react_grades_table.js.jsx.erb

```diff
@@ -53,23 +53,30 @@
 
   var GradeEntryBox = React.createClass({
     componentDidMount: function(){
-        this.props.value = this.props.default_value;
+      this.props.value = this.props.default_value;
     },
+
     handleChange: function(event) {
-      var updated_grade = event.target.value === '' ? null : parseFloat(event.target.value);
+      clearTimeout(this.typing_timer);
+
+      var updated_grade = event.target.value === '' ? null : Number(event.target.value);
       var params = {
         'updated_grade':       updated_grade,
         'student_id':          this.props.student_id,
         'grade_entry_item_id': this.props.grade_entry_column,
         'authenticity_token':  AUTH_TOKEN
       };
-      $.ajax({
-        url:  this.props.data_action,
-        data: params,
-        type: 'POST',
-        dataType: 'script',
-      });
-      this.props.value = parseFloat(event.target.value);
+
+      var data_action = this.props.data_action;
+      this.typing_timer = setTimeout(function() {
```

Comment:
It seems like this should only happen if `updated_grade` is not NaN? Ideally the cell would be set to red immediately, rather than making a request to the server that's going to fail.